raco/myrial/optimizer_tests.py

```diff
@@ -1003,3 +1004,75 @@ def test_projecting_join_maintains_partitioning(self):
         # (in general, info could be h($0) && h($2)
         self.assertEquals(pp.partitioning().hash_partitioned,
                           frozenset([AttIndex(0)]))
+
+    def test_no_shuffle_for_partitioned_distinct(self):
```

Comment:
What about the generic case where shuffles cannot be removed but the local aggregate operator (before the shuffle) can be pushed into SQL?